common/models/templates/src/trie-model.ts

```diff
@@ -100,13 +100,62 @@ class Traversal implements LexiconTraversal {
    */
   totalWeight: number;
 
-  constructor(root: Node, prefix: string, maxWeight: number) {
+  constructor(root: Node, prefix: string, totalWeight: number) {
     this.root = root;
     this.prefix = prefix;
-    this.totalWeight = maxWeight;
+    this.totalWeight = totalWeight;
+  }
+
+  child(char: USVString): LexiconTraversal | undefined {
+    /*
+      Note: would otherwise return the current instance if `char == ''`.  If
+      such a call is happening, it's probably indicative of an implementation
+      issue elsewhere - let's signal now in order to catch such stuff early.
+    */
+    if(char == '') {
+      return undefined;
+    }
```

Comment:
Do we want to log something like this?

Reply:
It's hard for the mobile apps to properly log console-logging statements made from the WebWorker.  I don't know that it's worth the trouble when we don't already have a pattern established for capturing said logs.